### PR TITLE
Fix geowidget bugs and crashes

### LINKED ISSUE #1914: [Geo-widget] Bug on adding and viewing point on a Map
**Describe the bug**
Ability to add points on a map and be able to click on them to view more details (i.e. Family profile)
1. The Location permission doesn't seem to work well on Android API 31 and above thus one cannot initiate the map module from the navigation drawer  - screenshot shared below
2.  On reloading the map the different added points on the map are not persisted once you reload the map view

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to the Navigation drawer 
2. Clip on the map module  - Maps
3. Click and add a point on a maps
4. Error described above will occur depending on the device and API level it's using

**Expected behaviour**

1. Launch the Map module from the navigation drawer 
5. You will get a map view that will have either prepopulated points on a map as markers or none if it's a 1st-time launch 
6. There will be a floating action button with a plus sign (+) to be able to add a point 
7. On click, you can be able to scroll and move the cursor to a point of interest, Once settled on a point you can add a family, click DONE
8. A registration form will be launched and you will be prompted to complete it 
9. Once done you should see the new point added on a map 
10. On relaunch of the map view, all previously added points should be visible and clickable to take you to the family profile 

**Screenshots**

1. Permission Issue on Android version 13  - There seems to be no way to enable storage permissions on this API level. For some reason, the location permission is enabled but no option for storage permissions 

**Android 8.1.0**

**Android 13**

**Smartphone (please complete the following information):**
 - Device: Pixel 4a Emulator 
 - Android version:  13

**Smartphone (please complete the following information):**
 - Device: Pixel 4a physical device 
 - Android version: 13
 - Build number: TQ1A.221205.011

3. Bug with adding/viewing points on a map 

https://user-images.githubusercontent.com/4540684/208388612-b66574bd-b9bd-42d7-91b9-599630ad8b0b.mp4

**Smartphone (please complete the following information):**
 - Device: Samsung Galaxy Tab A (2016) physical device 
 - Android version: 8.1.0
 - Build number: M1AJQ.P585JXS3CTL1

**Additional context**
1. Added the latest configs to the eCBIS Preview server so it may be prudent to update the local.properties configs to point to it and be able to reproduce the bug 
2. Pull from  `master` to update the geo-widget app variant with the latest changes with bug fixes and enhancements on Sync, Registers and profiles

### PR DESCRIPTION
**Pull request by ekigamba**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #1914 

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/geowidget/src/main/java/org/smartregister/fhircore/geowidget/screens/GeoWidgetViewModel.kt [hunk 1]
```diff
@@ -83,18 +85,22 @@ constructor(val defaultRepository: DefaultRepository, val dispatcherProvider: Di
...
+                characteristic.value is Reference &&
+                  characteristic.valueReference.reference.contains("Location")
+              }!!
+              .valueReference
+              .referenceElement
+              .idPart
+          )
+
+        familiesList.add(Pair(family, familyLocation))
+      } catch (ex: com.google.android.fhir.db.ResourceNotFoundException) {
```
Comment (ellykits) [inline android/geowidget/src/main/java/org/smartregister/fhircore/geowidget/screens/GeoWidgetViewModel.kt:101]: Add import for `ResourceNotFoundException`

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainActivity.kt [hunk 2]
```diff
@@ -123,7 +123,7 @@ open class AppMainActivity : BaseMultiLanguageActivity(), OnSyncListener, Questi
 
   override fun onSync(state: State) {
     Timber.i("Sync state received is $state")
-    when (state) {
+    /*when (state) {
```
Comment (ellykits) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainActivity.kt:126]: Revert commented code. Issue was fixed.

File: android/geowidget/src/main/java/org/smartregister/fhircore/geowidget/screens/GeoWidgetFragment.kt [hunk 3]
```diff
@@ -214,6 +214,10 @@ open class GeoWidgetFragment : Fragment(), Observer<FeatureCollection> {
       if (geometry is Point) {
         points.add(geometry)
       }
+    } ?: return
+
+    if (featureCollection.features()!!.size == 0) {
```
Comment (ellykits) [inline android/geowidget/src/main/java/org/smartregister/fhircore/geowidget/screens/GeoWidgetFragment.kt:219]: Do we expect `featureCollection.features()` to never be `null`? This code has a potential of throwing `NullPointerException`.

**Review by ellykits — DISMISSED** [review]

**Comment by ekigamba** [comment]
@ellykits Thanks for the early review. Can we avoid reviewing PRs/code until it's ready for review? It's counter-productive pointing out issues in a PR that is not complete eg. the commented out code and other workarounds allowed me to run the app and test it

File: android/geowidget/src/main/java/org/smartregister/fhircore/geowidget/screens/GeoWidgetFragment.kt (hunk 3 above)
Comment (ekigamba) [reply android/geowidget/src/main/java/org/smartregister/fhircore/geowidget/screens/GeoWidgetFragment.kt:220]: Nice catch! It shouldn't be possible but I'll check for that for some sanity

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainActivity.kt (hunk 2 above)
Comment (ekigamba) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainActivity.kt]: Fixed

File: android/geowidget/src/main/java/org/smartregister/fhircore/geowidget/screens/GeoWidgetViewModel.kt (hunk 1 above)
Comment (ekigamba) [reply android/geowidget/src/main/java/org/smartregister/fhircore/geowidget/screens/GeoWidgetViewModel.kt:101]: Fixed

File: android/geowidget/src/main/java/org/smartregister/fhircore/geowidget/screens/GeoWidgetFragment.kt (hunk 3 above)
Comment (ekigamba) [reply android/geowidget/src/main/java/org/smartregister/fhircore/geowidget/screens/GeoWidgetFragment.kt:220]: Line 218 checks for the null. Do you agree?

**Comment by ekigamba** [comment]
### Status
- The geowidget instrumentation tests are timing out possibly due to some dependencies update. This check is a current blocker
- The storage permissions for Android 11/13 need to be fixed in the Kujaku module

**Review by pld — APPROVED** [review]

**Comment by FikriMilano** [comment]
restoring the branch to investigate why androidTest is not reporting code coverage at the moment

**Pull request merged**